src/manage_sql_tls_certificates.c

```diff
@@ -1647,3 +1648,79 @@ add_tls_certificates_from_report_host (report_host_t report_host,
 
   return 0;
 }
+
+gchar *
+tls_certificate_origin_extra_xml (const char *origin_type,
```

Comment:
The idea of the code organisation is that the "manage" layer does the data stuff,
and the "gmp" layer does all the GMP stuff (i.e. anything XML).  So getting the data
here is good, but generating the XML should be done in the GMP layer.

Reply:
I was expecting more special SQL when I started this function but I agree that it makes more sense in the GMP layer file, so I changed it in dc7a201f203faf217c8e395c842123b0cc3f50a3.

Reply:
Now it feels like the data code in the function is in the wrong place, but it will have to do.

I think we should move away from using the resource types in the GMP layer.  So no report_t task_t etc and no find_* functions.  That's what's making me feel like this code is doing work on the data.